src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -247,12 +247,18 @@ internal void Update(double number)
                 case AggregationType.Histogram:
                     {
                         int i;
-                        for (i = 0; i < this.histogramBuckets.ExplicitBounds.Length; i++)
+                        if (double.IsNaN(number))
```

Comment:
Interesting... 🤔  makes me wonder if `NaN` should be an error case that we just log instead of recording.

Reply:
Ya, I'm not sure. I was trying to fix the Prometheus exporter test for `PrometheusSerializerTests.HistogramNaN` since I was trying to compare by using the `==` operator and also `double.NaN` is greater than all number, I didn't check the result of `double.NaN > double.Infinity` though 😅

Reply:
Yea, doesn't make sense to me that we should increment the largest bucket... why not the smallest? or even a random one?

Interestingly, I checked the Java SDK, and it turns out they too increment the last bucket. Wonder if this is just accidental because I don't see anything in the spec.

Though I looked at that prometheus test and noted that `NaN` is a valid value in the [prometheus format](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#comments-help-text-and-type-information).

Maybe something @reyang could shed some light on.

Reply:
The [SDK Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#numerical-limits-handling) gives certain flexibility to the SDK implementations for perf consideration.

Prometheus Exporter spec (which is still Experimental) hasn't covered `NaN` histogram yet, further clarifications are needed.

Reply:
If perf is what drives our decision then it would make sense to try and eliminate the `if (double.IsNaN(number))` check.

If we settle on using `Array.BinarySearch` then `~Array.BinarySearch(buckets, double.NaN)` equals 0. Seems it would be slightly more efficient to increment the first bucket in this case.

Reply:
Though, the bummer of aggregating a `NaN` value is that it not only messes up one of the bucket counts, it also clobbers the sum.

Reply:
> If perf is what drives our decision then it would make sense to try and eliminate the `if (double.IsNaN(number))` check.

+1

This is also called out in the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#numerical-limits-handling):

_It is unspecified how the SDK should handle the input limits. The SDK authors MAY **leverage/follow the language runtime behavior for better performance, rather than perform a check on each value coming from the API**._

Reply:
> Though, the bummer of aggregating a `NaN` value is that it not only messes up one of the bucket counts, it also clobbers the sum.

In addition to `NaN`, other values could result in `NaN` as well, for example (note that currently histogram does not support negative value):

```csharp
histogram.Record(double.PositiveInfinity);
histogram.Record(double.NegativeInfinity);
```